### PR TITLE
Add org.kde.PlatformInputContexts.MaliitSailfishOS

### PR DESCRIPTION
**Pull request by rinigus**
This is an extension for KDE runtime that provides access to Sailfish OS on-screen keyboard for Flatpak Qt applications.

The plugin is based on original Maliit framework with just an input context part of it used over here. As the build system was not optimized for such separation upstream and I would prefer to keep my changes to the minimum allowing me to follow upstream in the future, the extension is built using "simple" buildsystem.

While it is now made for 5.14 version, I'd like to add 5.12 as well. As its the first extension contributed by me, please advice how to do that. Maybe its supposed to be done later, though.

### CODE REVIEW

File: org.kde.PlatformInputContexts.MaliitSailfishOS.appdata.xml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="runtime">
+  <id>org.kde.PlatformInputContexts.MaliitSailfishOS</id>
+  <metadata_license>CC0-1.0</metadata_license>
+  <name>Maliit plugin for SailfishOS</name>
+  <summary>Keyboard plugin used in Sailfish OS for Flatpak applications</summary>
+  <description>
+    <p>Keyboard plugin for Sailfish OS that supports it's native input.</p>
```
Comment (nedrichards) [inline org.kde.PlatformInputContexts.MaliitSailfishOS.appdata.xml]: nitpick, "it's" should be "its" here, without the apostrophe.

**Comment by nedrichards** [comment]
@rinigus yes, you can only review one branch at a time, you'd have to create the other branch later on.

File: org.kde.PlatformInputContexts.MaliitSailfishOS.appdata.xml (hunk 1 above)
Comment (rinigus) [reply org.kde.PlatformInputContexts.MaliitSailfishOS.appdata.xml]: Thanks, fixed now.

**Comment by nedrichards** [comment]
@rinigus out of interest, have you spoken to the KDE runtime and Qt Flatpak/Portals team about this stuff? I'm very pro as I still have some residual affection for Maalit from my MeeGo days (aaaah, the Swype keyboard) but just want to make sure we're all synced up about the interfaces, releases etc.

**Comment by nedrichards** [comment]
bot, build org.kde.PlatformInputContexts.MaliitSailfishOS

**Comment by rinigus** [comment]
Sure, I filed an issue at KDE bugzilla and asked to introduce this type of extensions. The change with the extension type to make maliit possible was merged last night and hence we can do so.

I will test it in 1-3 hours and report back on how it works in practice.

As for keyboard, no Swipe anymore... Jolla probably couldn't afford it

**Comment by rinigus** [comment]
@nedrichards - just tested, works exactly as expected on device. So, on my side, it is ready for publishing.

File: org.kde.PlatformInputContexts.MaliitSailfishOS.yaml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    sources:
+      - type: git
+        url: https://github.com/sailfishos-flatpak/maliit-framework.git
+        branch: flatpak
+        commit: c1ce5ed850597cf7b0a1ea22cd6450bf4217092e
+  - name: appdata
+    buildsystem: simple
+    build-commands:
+      - mkdir -p ${FLATPAK_DEST}/share/appdata
+      - cp org.kde.PlatformInputContexts.MaliitSailfishOS.appdata.xml ${FLATPAK_DEST}/share/appdata
```
Comment (LeandroStanger) [inline org.kde.PlatformInputContexts.MaliitSailfishOS.yaml]: New appdata files should use metainfo instead. https://www.freedesktop.org/software/appstream/docs/chap-Quickstart.html

Comment (rinigus) [reply org.kde.PlatformInputContexts.MaliitSailfishOS.yaml]: Can we skip appdata then?

Comment (rinigus) [reply org.kde.PlatformInputContexts.MaliitSailfishOS.yaml]: Can we skip appdata then?

Comment (rinigus) [reply org.kde.PlatformInputContexts.MaliitSailfishOS.yaml]: As there is a silence regarding skipping appdata - any example regarding it? Do I just rename it and mv to $PREFIX/share/metainfo/%{id}.metainfo.xml?

**Comment by rinigus** [comment]
Changes applied, please review

**Comment by nedrichards** [comment]
bot, build org.kde.PlatformInputContexts.MaliitSailfishOS

**Comment by rinigus** [comment]
Thank you very much!

**Pull request closed without merging**